### PR TITLE
Module os_network_free_fip to discover free floating IP and some refactoring

### PR DESCRIPTION
**Pull request by brtkwr**
Closes: https://github.com/cedadev/jasmin-cluster-as-a-service/issues/35

### CODE REVIEW

File: group_vars/openstack [hunk 1]
```diff
@@ -17,17 +17,19 @@ openstack_auth:
   auth_url: "{{ lookup('env', 'OS_AUTH_URL') }}"
 
 cluster_image: centos-7-20190104
+cluster_floating_network_id: NERCvSE
+
 cluster_deploy_user: centos
 cluster_delegate_role: _member_
-cluster_gw_group: gw
+cluster_gw_group: ephemeral
```
Comment (oneswig) [inline group_vars/openstack:24]: Why the change in name?

**Review by oneswig — APPROVED** [review]
Nice to have so many pieces of work in one PR :-)

File: group_vars/openstack (hunk 1 above)
Comment (brtkwr) [reply group_vars/openstack:24]: Makes more sense than gw?

Comment (mkjpryor-stfc) [reply group_vars/openstack:24]: I’m going to be really obtuse and say `ephemeral-gw` makes most sense... :P

Comment (mkjpryor-stfc) [reply group_vars/openstack:24]: Or even `deploy-gw` if we want it to really say what it does.
